src/application-insights/azext_applicationinsights/_validators.py

```diff
@@ -8,6 +8,14 @@
 from msrestazure.tools import is_valid_resource_id
```

Comment:
>is_valid_resource_id [](start = 30, length = 20)

use from azure.mgmt.core.tools import parse_resource_id, is_valid_resource_id instead, we'll deprecate msrestazure.tools in the future.